src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,391 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import itertools
+
+import numpy as np
+
+from pymor.core.base import BasicObject
+from pymor.models.transfer_function import TransferFunction
+
+
+class pAAAReductor(BasicObject):
```

Comment:
Can p-AAA support parametric stationary problems (`StationaryModel`)? If yes, do we want to have this in this PR?

Reply:
Thinking about it more, one big question here would be: What would the ROM look like for a stationary model? Essentially, p-AAA would approximate the inverse of the parametric operator via a rational function. I feel like I would have to define an operator where the `apply_inverse` function corresponds to evaluating a rational function and then the ROM is just a stationary model with this operator construction as the `operator` attribute. Or is there an easier way that I am missing? If not maybe it's a bit more involved and we should work on this in a separate PR.

Reply:
> The idea is to treat the solutions of a `StationaryModel` as MIMO input-output data.

I guess you mean the outputs of a `StationaryModel`, and solutions if there is no output?

> Thinking about it more, one big question here would be: What would the ROM look like for a stationary model?

I see now that the reductor is returning a `TransferFunction`. For `StationaryModels`, maybe it could return a specialized subclass of `Model` that evaluates the rational function when its `output` method is called. Maybe `TransferFunction` itself should be a `Model` with `s` as a parameter... But, this would involve a discussion about the model hierarchy, which might be better to do some other time (@sdrave).

Reply:
> I guess you mean the outputs of a `StationaryModel`, and solutions if there is no output?

That's right.

> I see now that the reductor is returning a `TransferFunction`. For `StationaryModels`, maybe it could return a specialized subclass of `Model` that evaluates the rational function when its `output` method is called. Maybe `TransferFunction` itself should be a `Model` with `s` as a parameter... But, this would involve a discussion about the model hierarchy, which might be better to do some other time (@sdrave).

Yes, it's really not so clear. p-AAA can be used to approximate any (multivariate) function, so the "FOM" and thus the "ROM" could just be functions. I agree, we should defer this until there is a model hierarchy discussion.